### ISSUE TITLE
:sparkles: Allow anyone to create an application

### ISSUE THREAD

**Issue by micedre** [InseeFr/sugoi-api#675]
As a sugoi user, I would like to be able to create an application and add group (and user to this application). At the moment, application creation is reserved to realm admins, so I have to ask them to create the application and give me rights to it.

This should be configurable by realm, and if enabled, any reader on the realm should have the rights to create application. When application created, the user should have all rights on it.

The user creating application should also have a right to delete it.

**Comment by CChemin** [comment]
I don't think this is the good way to do this. There is no reason that a reader should have writing rights and we might have some use cases where only a few people is allowed to create application and not all readers.

To do so I really prefer to add a new role (like application_creator_{realm}) and users with this role will be able to create any application + with this role by default on all authenticated user we have more or less the same feature (the reader role will not be needed).

**Comment by emmanuelbar** [comment]
Regarding the deletion of the application: that would work with a new role (like "applicationX_appmanager" ) ?
